--- hiredis-0.11.0/Makefile.orig	2013-12-12 17:28:25.817204756 +0100
+++ hiredis-0.11.0/Makefile	2013-12-12 17:33:18.150525821 +0100
@@ -76,8 +76,8 @@
 	$(CC) -o $@ $(REAL_CFLAGS) $(REAL_LDFLAGS) -I$(AE_DIR) $(AE_DIR)/ae.o $(AE_DIR)/zmalloc.o example-ae.c $(STLIBNAME)
 endif
 
-hiredis-%: %.o $(STLIBNAME)
-	$(CC) -o $@ $(REAL_LDFLAGS) $< $(STLIBNAME)
+hiredis-%: %.o $(DYLIBNAME)
+	$(CC) -o $@ $(REAL_LDFLAGS) $< -l$(subst lib,,$(LIBNAME))
 
 test: hiredis-test
 	./hiredis-test
